apps/balancer-tools/src/components/Slider.tsx

```diff
@@ -0,0 +1,118 @@
+}
+
+const BaseSlider = React.forwardRef<
+  React.ElementRef<typeof SliderPrimitive.Root>,
+  Omit<ISlider, "label" | "unit">
+>(({ className, min, max, defaultValue, name, ...props }, ref) => {
+  const minNumber = Number(min);
+  const maxNumber = Number(max);
+  const { getValues, setValue } = useFormContext();
+  const meanValue = (minNumber + maxNumber) / 2;
+  const value = name ? getValues(name) || meanValue : meanValue;
+
+  return (
+    <SliderPrimitive.Root
+      ref={ref}
+      className={cn(
+        "relative flex items-center select-none touch-none w-[200px] h-8 my-4 w-full",
+        className,
+      )}
+      min={minNumber}
+      max={maxNumber}
+      defaultValue={[defaultValue || meanValue]}
+      onValueChange={(value) => setValue(name, value[0])}
+      value={[value ? Number(value) : meanValue]}
+      {...props}
+    >
+      <SliderPrimitive.Track className="bg-blue4 relative grow rounded-full h-[3px]">
+        <SliderPrimitive.Range className="absolute bg-blue7 rounded-full h-full" />
+      </SliderPrimitive.Track>
+      <SliderPrimitive.Thumb
+        className="block w-5 h-5 bg-blue7 shadow-[0_2px_10px] shadow-blackA7 rounded-[10px] hover:bg-violet3 focus:outline-none focus:shadow-[0_0_0_5px] focus:shadow-blackA8"
+        aria-label="slider-thumb"
+      >
+        <h1 className="absolute ease-in-out duration-150 opacity-100 -translate-y-8 -translate-x-6 px-5 py-1 bg-blue7 text-center rounded-full text-white whitespace-nowrap text-xs font-bold">
+          {value}
+        </h1>
+      </SliderPrimitive.Thumb>
+    </SliderPrimitive.Root>
+  );
+});
+
+BaseSlider.displayName = SliderPrimitive.Root.displayName;
+
+export const Slider = React.forwardRef<HTMLInputElement, ISlider>(
+  ({
+    name,
+    label,
+    validation,
+    tooltipText,
+    tooltipLink,
+    unit,
+    ...props
+  }: ISlider) => {
+    const {
+      register,
+      formState: { errors },
+    } = useFormContext();
+
+    if (!name) {
+      throw new Error("Input component requires a name prop");
+    }
+
+    const error = errors[name] as FieldError | undefined;
+    const errorMessage = error?.message;
+
+    return (
+      <div className="flex flex-col">
+        <div className="flex flex-row justify-between">
+          <FormLabel className="mb-2 block text-sm text-slate12">
+            {label} {unit ? `(${unit})` : ""}
+          </FormLabel>
+          {tooltipText && tooltipLink && (
+            <Tooltip content={tooltipText}>
+              <a href={tooltipLink} target="_blank">
+                <InfoCircledIcon color={slateDarkA.slateA11} />
+              </a>
+            </Tooltip>
+          )}
+        </div>
+        <FormControl>
+          <BaseSlider {...props} {...register(name, validation)} />
+        </FormControl>
+        {errorMessage && (
+          <FormMessage className="mt-1 h-6 text-sm text-tomato10">
+            <span>{errorMessage}</span>
+          </FormMessage>
+        )}
+      </div>
```

Comment:
This is a lot of repetition form input, right? You could have refactored the Input to have a e.g. BaseFormElement that has all this and gets children with the element implementation